elide-async/src/main/java/com/yahoo/elide/async/service/AsyncExecutorService.java

```diff
@@ -133,10 +127,12 @@ public void executeQuery(AsyncQuery queryObj, User user, String apiVersion) {
      * @param user User
      * @param apiVersion API Version
      */
-    public void completeQuery(AsyncQuery query, User user, String apiVersion) {
-        if (query.getQueryUpdateWorker() != null) {
+    public void completeQuery(AsyncAPI query, User user, String apiVersion) {
+        AsyncAPIUpdateThread queryUpdateWorker = futureTask.get();
+        if (queryUpdateWorker != null) {
             log.debug("Task has not completed");
-            updater.execute(query.getQueryUpdateWorker());
+            updater.execute(queryUpdateWorker);
+            futureTask.remove();
```

Comment:
This could be missed because of an exception - and then Elide will remain in an invalid state.

Instead of using null/non-null - you'll need to keep track of something inside the threadlocal itself.